seatunnel-connectors-v2/connector-file/connector-file-cos/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/cos/source/CosFileSourceFactory.java

```diff
@@ -97,4 +102,14 @@ public OptionRule optionRule() {
     public Class<? extends SeaTunnelSource> getSourceClass() {
         return CosFileSource.class;
     }
+
+    @Override
+    public <T, SplitT extends SourceSplit, StateT extends Serializable>
+            TableSource<T, SplitT, StateT> createSource(TableSourceFactoryContext context) {
+        return () ->
+                (SeaTunnelSource<T, SplitT, StateT>)
+                        new CosFileSource(
+                                context.getOptions(),
+                                CatalogTableUtil.buildWithConfig(context.getOptions()));
```

Comment:
-1. We should build catalogtable with file sometimes. Please refer https://github.com/apache/seatunnel/pull/9121/files#diff-1f7ebddd2239c9ee909ee66c8f67c119b92f05324519ea5ce40d5aeab98da984L124